### PR TITLE
[core] update l10n issue with a single command line

### PR DESCRIPTION
**Pull request by alexfauquette**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -167,6 +167,7 @@
     ]
   },
   "dependencies": {
+    "@octokit/rest": "^18.12.0",
```
Comment (m4theushw) [inline package.json:170]: It's a `devDependency`.

**Comment by m4theushw** [comment]
Do you plan to create the GitHub workflow? It would be nice to have this automated.

**Comment by alexfauquette** [comment]
I'm not planning to do it for now, because I do not know exactly how files are managed in github workflow. Which could be a problem, especially with the following part of the code.

```ts
function git(args: any) {
  return new Promise((resolve, reject) => {
    exec(`git ${args}`, (err, stdout) => {
      if (err) {
        reject(err);
      } else {
        resolve(stdout.trim());
      }
    });
  });
}
```
I agree it would be better, but for now, updating the issue at least once a week during the release seems enought

**Review by m4theushw — APPROVED** [review]

File: scripts/README.md [hunk 2]
```diff
@@ -63,6 +63,16 @@ You can use the following script in your browser console on any GitHub page to a
 4. Release the versions on NPM: `yarn release:publish` (you need your 2FA device).
 5. Create a new tag named with the release you just did `git tag v4.0.0-alpha.30 && git push upstream --tag`
 
+### Update l10n issue
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply scripts/README.md:66]: So I let it, and you will remove it in the PR to add the GitHub workflow?

**Comment by m4theushw** [comment]
> I'm not planning to do it for now, because I do not know exactly how files are managed in github workflow. Which could be a problem, especially with the following part of the code.

It works in the same way as in your machine. We first `git checkout` with a special GitHub action, then `yarn install` and finally run the command to update the issue.

https://github.com/mui-org/material-ui/blob/master/.github/workflows/ci.yml#L22

**Comment by alexfauquette** [comment]
@m4theushw I made the script to make the update when there is a push on master. If you can verify it. I don't know how to test it except merging and hope it works

File: .github/workflows/l10n.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: CI
```
Comment (m4theushw) [inline .github/workflows/l10n.yml:1]:
```suggestion
name: Update missing translations
```

File: .github/workflows/l10n.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          cache: 'yarn' # https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-dependencies
+      - run: yarn install
+        env:
+          # Don't need playwright in this job
+          PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: yarn release:tag
```
Comment (m4theushw) [inline .github/workflows/l10n.yml:26]:
```suggestion
      - name: yarn l10n --report
```

File: .github/workflows/l10n.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        run: |
+          git remote -v
+          yarn l10n --report
```
Comment (m4theushw) [inline .github/workflows/l10n.yml:29]: The `GITHUB_TOKEN` envvar is missing.

**Comment by m4theushw** [comment]
> I made the script to make the update when there is a push on master. If you can verify it. I don't know how to test it except merging and hope it works

It's a shot in the dark, we test after merge. You could configure GitHub Actions in your fork. It might not work initially but we can fix.

**Comment by alexfauquette** [comment]
Bot is live 🤖🎉

image

**Pull request merged**